### DISCUSSION TITLE
Production Deployment: Build Locally or on Production Server?

### DISCUSSION THREAD

**Discussion by mkastner** [keystonejs/keystone#8146] [Questions] [answered]
I am wondering, what's the recommended or maybe even best practice way for building.

When building, will any binary dependencies need to be compiled and linked?

The reason why I am asking is this: I am developing on MacOS and the production deployment target is FreeBSD. Therefor the situation is a little different from old fashioned Docker/K8S deployment.

My preferred way would be to build locally but if that's not possible I guess I need to build on the target machine.

**Comment by flexdinesh** [comment, marked as answer]
You will need to build on target machine since Keystone uses Prisma internally and prisma binaries and generated during build based on target machine. [You can read more about prisma binaries here](https://www.prisma.io/docs/concepts/components/prisma-engines#prisma-engines).

**Comment by mkastner** [comment]
@flexdinesh Thanks very much for your clarification, and my apologies for the late reply.
